securedrop/tests/functional/sd_config_v2.py

```diff
@@ -0,0 +1,102 @@
+from dataclasses import dataclass, field
```

Comment:
This file contains a proposal for tweaking the SDConfig object. It is only used in the test modified in this PR right now.

Reply:
Note that this new `SecureDropConfig` works transparently as a replacement for `SDConfig`, because it has the same attributes.

Reply:
I really like the idea of using a frozen dataclass for config, it makes perfect sense. But I'm wondering if it would make more sense to just do this in SDConfig itself, without trying to maintain a parallel implementation for tests.

We could add a proper `__init__` to SDConfig and move the current one to be a classmethod `from_config()` or something. Then we would swap SDConfig into a dataclass underneath that while still supporting the pretty permissive creation style we have currently.

Reply:
The goal is for this v2 to eventually replace SDConfig. I didn't do it here because it would require a lot more changes: importing sdconfig.py requires a config.py to be present, and changing this requires a bigger refactoring (which I plan on doing eventually). 
Also based on previous PRs, touching SDConfig has the potential to break compatibility with running SD instances and should be done separately, in a small PR.